### PR TITLE
Dzollo dev config fix

### PR DESCRIPTION
**Pull request by denniszollo**
/cc @jacobmcnamee @mfine
Goal is to fail loudly if we ever are unable to configure the FPGA or RT firmware whilst in DEV mode.  Some drive tests showed up with the wrong FPGA bitstream.

Still needs to even be sniff tested and sanity checked.  This could easily have a typo or something silly.

### CODE REVIEW

**Comment by cbeighley** [comment]
@jacobmcnamee to review

File: board/piksiv3/rootfs-overlay/etc/init.d/S84piksi_fpga [hunk 1]
```diff
@@ -8,7 +8,11 @@ start() {
     cat "$bit" > /dev/xdevcfg
     prog_done=`cat /sys/devices/soc0/amba/f8007000.devcfg/prog_done`
     if [ "$prog_done" != "1" ]; then
-      echo "ERROR: configuration failed"
+      while [ 1 ]; do
```
Comment (jacobmcnamee) [inline board/piksiv3/rootfs-overlay/etc/init.d/S84piksi_fpga]: The firmware already loops waiting for the FPGA if it is not configured. I recommend not blocking here.

File: board/piksiv3/rootfs-overlay/etc/init.d/S75dev_boot [hunk 2]
```diff
@@ -33,7 +37,11 @@ copy_from_net() {
   tftp -g -r "$net_file" -l "$output_file" -b 65464 "$server_ip"
 
   if [ ! -f "$output_file" ]; then
-    echo "ERROR: copy failed"
+    while [ 1 ]; do
+      echo "ERROR: copy of images from SD card failed. Please Reboot"
```
Comment (jacobmcnamee) [inline board/piksiv3/rootfs-overlay/etc/init.d/S75dev_boot]: "from network"

File: board/piksiv3/rootfs-overlay/etc/init.d/S75dev_boot [hunk 3]
```diff
@@ -16,7 +16,11 @@ copy_from_sd() {
   cp "$sd_file" "$output_file"
 
   if [ ! -f "$output_file" ]; then
-    echo "ERROR: copy failed"
+    while [ 1 ]; do
+      echo "ERROR: copy of images from SD card failed. Please Reboot"
+      echo "ERROR: copy of images from SD card failed. Please Reboot" | SBP_LOG --error
```
Comment (jacobmcnamee) [inline board/piksiv3/rootfs-overlay/etc/init.d/S75dev_boot]: ZMQ infrastructure will not be up when this script executes, so the SBP log stuff will not be visible and possibly hang.

**Comment by jacobmcnamee** [comment]
I think it might be better to check if the FPGA is already configured and reset it if so. Let me try a couple things and get back to you with steps for doing that. It should be pretty simple.

**Comment by jacobmcnamee** [comment]
Alright, I'm abandoning my idea of resetting the FPGA. The reset part works, but I've seen some strange errors afterward and reconfiguring has not yet worked. It seems that the Xilinx `xdevcfg` driver is buggy.

I suggest keeping the original strategy of blocking with an error message if the file copy fails.

**Comment by anonymous-travis-swiftnav** [comment]
## v1.0.0-branch-113-g91b188c
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-113-g91b188c/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.0.0-branch-113-g91b188c/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-113-g91b188c

**Pull request closed without merging**